src/meteors/lime.py

```diff
@@ -241,156 +252,453 @@ def to(self, device: torch.device) -> Self:
+            band_ranges_indices (Dict[str, Iterable[Tuple[int, int]] | Tuple[int, int]]): dictionary containing user-made band ranges as keys and tuple of start and end indices as values. For instance it could equal to `{"A": (0, 4), "B": [(6, 9), (20, 23)]}`. "A" and "B" would be the names of the segments, the created segment indices would equal to [0, 1, 2, 3, 4] and [6, 7, 8, 20, 21, 22]
+            band_ranges_wavelengths (Dict[str, list[Tuple[float, float]] | Tuple[float, float]]): dictionary containing user-made band ranges as keys and tuple of start and end wavelengths as values. For instance, if our image contains wavelengths [400, 405, 410, 420, 450], then dictionary could equal to `{"A": (400, 410), "B": [(420, 450)]}`. "A" and "B" would be the names of the segments, the created segment indices would equal to [0, 1, 2] and [3, 4]
+
+            device (torch.device | str | None): a device on which the band mask will be created. If None, it is created on the same device as the image. Defaults to None
+            repeat_dimensions bool: Whether to repeat the band mask over all image dimensions For instance, if False the output mask could have shape (1, 1, 150). If True - (64, 64, 150). Defaults to False
+
 
         Returns:
             Tuple[torch.Tensor, Dict[str, int]]: a tuple which consists of an actual band mask and a dictionary that translates the band names into the segment values
         """
 
-        if isinstance(band_names, dict):
-            band_names_dict = {tuple(k) if not isinstance(k, str) else k: v for k, v in band_names.items()}
-        else:
-            band_names_dict = Lime.__get_band_dict_from_list(band_names)  # type: ignore
+        if not isinstance(image, Image):
+            raise ValueError("Image should be an instance of Image class")
+
+        # validate types
+
+        dict_labels_to_segment_ids = None
+        if band_names is not None:
+            logger.debug("Getting band mask from band names of spectral bands")
+            if not isinstance(band_names, (list, dict)):
+                raise TypeError("Band names should be either a list or a dictionary")
+            if isinstance(band_names, dict):
+                if not all(isinstance(key, (tuple, str)) for key in band_names.keys()):
+                    raise TypeError("Keys in band names dictionary should be either a tuple or a string")
+                if not all(isinstance(value, int) for value in band_names.values()):
+                    raise TypeError("Values in band names dictionary should be integers")
+            if isinstance(band_names, list):
+                if not all(isinstance(segment, (str, list)) for segment in band_names):
+                    raise TypeError("Elements in band names list should be either a string or a list")
+
+            band_ranges_wavelengths, dict_labels_to_segment_ids = Lime._get_band_ranges_wavelengths_from_band_names(
```

Comment:
Mypy gives me:
`src/meteors/lime.py:316: error: Incompatible types in assignment (expression has type "dict[Union[str, tuple[str, ...]], list[tuple[float, float]]]", variable has type "Optional[dict[Union[str, tuple[str, ...]], Union[Iterable[tuple[float, float]], tuple[float, float]]]]")  [assignment]
Found 1 error in 1 file (checked 12 source files)`

Reply:
ignoring as @Fersoil decided